internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -310,6 +317,14 @@ func resourceKustoClusterCreateUpdate(d *pluginsdk.ResourceData, meta interface{
 		TrustedExternalTenants: expandTrustedExternalTenants(d.Get("trusted_external_tenants").([]interface{})),
 	}
 
+	if v, ok := d.GetOk("public_ip_type"); ok {
```

Comment:
Since it has default value, you can always get its value. This part can be simplified like the following:
```
clusterProperties.PublicIPType = kusto.PublicIPType(d.Get("public_ip_type").(string))
```
And we can add this field when declare `clusterProperties `

Reply:
Done